Add EDOT Node.js docs for ELASTIC_OTEL_CONTEXT_PROPAGATION_ONLY and send_metrcs et al central config settings (#412)
elastic/opentelemetry: docs/reference/edot-sdks/nodejs/configuration.md

| `OTEL_EXPORTER_OTLP_METRICS_TEMPORALITY_PREFERENCE` 🔹 | [(EDOT Ref)](#otel_exporter_otlp_metrics_temporality_preference-details) The metrics exporter's default aggregation `temporality`. The default value is `delta`. The upstream OTel default is `cumulative`. |
| | |
| `OTEL_SEMCONV_STABILITY_OPT_IN` 🔹 | [(EDOT Ref)](#otel_semconv_stability_opt_in-details) Control which HTTP semantic conventions are used by `@opentelemetry/instrumentation-http`. The default value is `http`. The upstream OTel default is an empty value. |

> | `ELASTIC_OTEL_CONTEXT_PROPAGATION_ONLY` 🔹 | [(EDOT Ref)](#elastic_otel_context_propagation_only-details) Set to `true` to enable trace-context propagation in outgoing requests and log correlation, but disable the sending of spans. |

Reviewer: ```suggestion
| `ELASTIC_OTEL_CONTEXT_PROPAGATION_ONLY` 🔹 | [(EDOT Ref)](#elastic_otel_context_propagation_only-details) Set to `true` to turn on trace-context propagation in outgoing requests and log correlation. This turns off the sending of spans. |
```

| Logging level | `logging_level` | Dynamic |
| Turn off instrumentations | `deactivate_instrumentations` | Dynamic |
| Turn off all instrumentations | `deactivate_all_instrumentations` | Dynamic |
| Send traces | `send_traces` | Dynamic {applies_to}`product: preview 1.3.0` |
| Send metrics | `send_metrics` | Dynamic {applies_to}`product: preview 1.3.0` |

> | Send logs | `send_logs` | Dynamic {applies_to}`product: preview 1.3.0` |

Reviewer: Excellent usage of inline applies_to!

EDOT Node.js differs from current upstream OTel JS in that it *defaults `OTEL_SEMCONV_STABILITY_OPT_IN` to `http`*. This means that, by default, all HTTP-related telemetry from EDOT Node.js will use the newer, stable HTTP semantic conventions. (This difference from upstream is expected to be temporary, as upstream `@opentelemetry/instrumentation-http` switches to producing only stable HTTP semantic conventions after its transition period.)

### `ELASTIC_OTEL_CONTEXT_PROPAGATION_ONLY` details [elastic_otel_context_propagation_only-details]

```{applies_to}
product:
  edot_node: preview 1.3.0
```

> Set `ELASTIC_OTEL_CONTEXT_PROPAGATION_ONLY` to `true` to enable trace-context propagation in outgoing requests and log correlation, but disable the sending of spans.
> Setting `OTEL_TRACES_EXPORTER=none` will "win" over this setting: trace-context propagation will be disabled.
> 
> See the [migration guide](/reference/edot-sdks/nodejs/migration.md#contextpropagationonly) for details on how this relates to the similar `contextPropagationOnly` setting from the non-OTel Elastic Node.js APM agent.

Reviewer: ```suggestion
Set the `ELASTIC_OTEL_CONTEXT_PROPAGATION_ONLY` environment variable to `true` to turn on trace-context propagation in outgoing requests and log correlation. This turns off the sending of spans. Setting `OTEL_TRACES_EXPORTER` to `none` overrides this setting, deactivating trace-context propagation.

Refer to the [migration guide](/reference/edot-sdks/nodejs/migration.md#contextpropagationonly) for details on how this relates to the similar `contextPropagationOnly` setting from the non-OTel Elastic Node.js APM agent.
```
